cmd/dep/init.go

```diff
@@ -164,8 +175,8 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	}
 	p.Lock = dep.LockFromSolution(soln)
 
-	rootAnalyzer.FinalizeRootManifestAndLock(p.Manifest, p.Lock)
-	gs.FinalizeRootManifestAndLock(p.Manifest, p.Lock)
+	rootAnalyzer.RemoveTransitiveDependencies(p.Manifest)
```

Comment:
@carolynvs  I used `RemoveTransitiveDependencies()` in this PR itself. Hopefully this is how you intended to use it 😅